service/start.go

```diff
@@ -119,6 +115,132 @@ func (svc *service) startNostr(ctx context.Context, encryptionKey string) error
 	return nil
 }
 
+type createAppSubscriber struct {
+	events.EventSubscriber
+	svc   *service
+	relay *nostr.Relay
+}
+
+func (s *createAppSubscriber) ConsumeEvent(ctx context.Context, event *events.Event, globalProperties map[string]interface{}) {
+	switch event.Event {
+	case "app_created":
+		properties, ok := event.Properties.(map[string]interface{})
+		if !ok {
+			logger.Logger.WithField("event", event).Error("Failed to cast event.Properties to map")
+			return
+		}
+		walletChildPubkey, ok := properties["walletChildPubkey"].(string)
+		if !ok {
+			logger.Logger.WithField("event", event).Error("Failed to get app walletChildPubkey")
+			return
+		}
+		id, ok := properties["id"].(uint)
+		if !ok {
+			logger.Logger.WithField("event", event).Error("Failed to get app id")
+			return
+		}
+		if walletChildPubkey != "" {
+			go func() {
+				appWalletPrivKey, err := s.svc.keys.GetBIP32ChildKey(uint32(id))
+				if err != nil {
+					logger.Logger.WithError(err).Error("Failed to calculate app wallet priv key")
+				}
+				err = s.svc.startAppWalletSubscription(ctx, s.relay, walletChildPubkey, appWalletPrivKey)
+				if err != nil {
+					logger.Logger.WithError(err).Error("Failed to subscribe to wallet")
+				}
+				// TODO what should we do?
+			}()
+		}
+	}
+}
+
+type deleteAppSubscriber struct {
+	events.EventSubscriber
+	walletChildPubkey string
+	relay             *nostr.Relay
+	nostrSubscription *nostr.Subscription
+	svc               *service
+	infoEventId       string
+}
+
+func (s *deleteAppSubscriber) ConsumeEvent(ctx context.Context, event *events.Event, globalProperties map[string]interface{}) {
+	switch event.Event {
+	case "app_deleted":
+		properties, ok := event.Properties.(map[string]interface{})
+		if !ok {
+			logger.Logger.WithField("event", event).Error("Failed to cast event.Properties to map")
+			return
+		}
+
+		walletChildPubkey, ok := properties["walletChildPubkey"].(string)
+		if s.walletChildPubkey == walletChildPubkey {
+			id, _ := properties["id"].(uint)
+			s.nostrSubscription.Unsub()
+			appWalletPrivKey, _ := s.svc.keys.GetBIP32ChildKey(uint32(id))
+			err := s.svc.nip47Service.PublishNip47InfoDeletion(ctx, s.relay, walletChildPubkey, appWalletPrivKey, s.infoEventId)
+			if err != nil {
+				logger.Logger.WithField("event", event).Error("Failed to publish nip47 info deletion")
+			}
+		}
+	}
+}
+
+func (svc *service) startAllExistingAppsWalletSubscriptions(ctx context.Context, relay *nostr.Relay) {
+	var apps []db.App
+	result := svc.db.Where("wallet_child_pubkey != ?", "").Find(&apps)
```

Comment:
is this query correct, shouldn't it be `wallet_child_pubkey IS NOT NIL`?

Reply:
I'm actually not sure, because if for some reason an app DB row is created but then it fails to update the column with the calculated pubkey it will have an empty string '' as value, which the current query filters, and `IS NOT NULL` does not.
Is there a way to make sure that when a new DB row is created without passing a value to that column it will have NULL value instead of ''?

Reply:
Ah, gorm uses the default value of the column type. If it was a `*string` instead of a `string` it would work. But maybe we will ensure all the apps have keys, so this is no longer an issue.

Reply:
@frnandu I think this would be a good change and actually to do the rename now (https://github.com/getAlby/hub/issues/744). I will make this change